### PR TITLE
Move OpenStack inventory.py out of sample-inventory

### PR DESCRIPTION
**Pull request by tomassedovic**
This moves the dynamic inventory for OpenStack from the `sample-inventory` dir users are expected to copy and edit to `/inventory/openstack`.

This will ensure that people don't use an outdated inventory as well as communicates that the inventory (similar to the playbooks and roles) is not something they are expected to modify under normal circumstances.

The sample inventory will now contain only `group_vars`  and as such, moving from that to providing vars explicitly is also more obvious (you replace the `-i inventory` command with `-e @vars.yaml` and you're good to go).

### CODE REVIEW

**Comment by tomassedovic** [comment]
/cc @mbruzek @jmencak @markllama @tzumainn @celebdor @luis5tb @bogdando 

(sorry for the spam, but this changes the way you deploy on openstack so I wanted to bring it to your attention)

File: playbooks/openstack/README.md [hunk 1]
```diff
@@ -183,14 +183,21 @@ Then run the provision + install playbook -- this will create the OpenStack
 resources:
 
 ```bash
-$ ansible-playbook --user openshift -i inventory \
+$ ansible-playbook --user openshift \
+  -i openshift-ansible/inventory/openstack/inventory.py
+  -i inventory \
```
Comment (bogdando) [inline playbooks/openstack/README.md]: I didn't know multiple `-i` work!

Comment (tomassedovic) [reply playbooks/openstack/README.md]: Yeah. We even mention it in the readme. It's pretty useful :-).

**Comment by tomassedovic** [comment]
Ah, these failures seem genuine. Something to do with building the release.

**Comment by tomassedovic** [comment]
@michaelgugino @sdodson What would be a good location for dynamic inventories?

The top-level `inventory` directory seems to be for examples only (indicated by the contents as well as the RPM script which copies them under docs). I've put it to `playbooks/openstack/inventory.py` for now, but maybe you have a better idea?

**Comment by bogdando** [comment]
@sdodson would you please LGTM?

**Pull request merged**